src/returnModeUtils.h

```diff
@@ -14,9 +14,42 @@
 
 #pragma once
 
+#include <cassert>
+
+#include <macros/cuda.hpp>
 #include <rgl/api/core.h>
 
-inline size_t getReturnCount(rgl_return_mode_t returnMode)
+inline HostDevFn size_t getReturnCount(rgl_return_mode_t returnMode)
 {
 	return static_cast<size_t>(returnMode >> RGL_RETURN_MODE_BIT_SHIFT);
 }
+
+inline HostDevFn rgl_return_type_t getReturnType(rgl_return_mode_t returnMode, size_t returnIndex)
+{
+	return static_cast<rgl_return_type_t>((returnMode >> (returnIndex * RGL_RETURN_TYPE_BIT_SHIFT)) & 0xff);
+}
+
+inline HostFn const std::string& getReturnTypeName(rgl_return_type_t returnType)
+{
+	// clang-format off
+	static const std::unordered_map<rgl_return_type_t, std::string> typeNames = {
+		{RGL_RETURN_TYPE_UNKNOWN, "Unknown"},
+		{RGL_RETURN_TYPE_FIRST, "First"},
+		{RGL_RETURN_TYPE_SECOND, "Second"},
+		{RGL_RETURN_TYPE_LAST, "Last"},
+	    {RGL_RETURN_TYPE_STRONGEST, "Strongest"},
+	    {RGL_RETURN_TYPE_SECOND_STRONGEST, "SecondStrongest"},
+	};
+	// clang-format on
+	assert(typeNames.find(returnType) != typeNames.cend());
+	return typeNames.at(returnType);
```

Comment:
`switch` may be cleaner for this, but I am not forcing it.

Reply:
I will stay with a map here, that on the other hand feels cleaner for me. And I see no downgrade of using map.